core/cas-server-core-cookie-api/src/main/java/org/apereo/cas/web/support/mgmr/DefaultCasCookieValueManager.java

```diff
@@ -56,6 +56,11 @@ protected String buildCompoundCookieValue(final String givenCookieValue, final H
         return builder.toString();
     }
 
+
+    /**
+     * Make sure cookie is used from same IP and with same user-agent as when cookie created.
+     * Client info (with original client ip) may be null if cluster failover occurs and session replication not working.
+     */
```

Comment:
Would be best to move these up to the class level. Not sure how Override works with javadocs in general.